lib/Driver/ToolChains.cpp

```diff
@@ -1115,9 +1115,16 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
   // Add the linker script that coalesces protocol conformance sections.
   Arguments.push_back("-Xlinker");
   Arguments.push_back("-T");
-  Arguments.push_back(
+    
+  // FIXME: This should also query the abi type (i.e. gnueabihf)
+  if (getTriple().getArch() == llvm::Triple::arm) {
+    Arguments.push_back(
+      context.Args.MakeArgString(Twine(RuntimeLibPath) + "/armv7/swift.ld"));
```

Comment:
Why not just concatenate in `getTriple().getArchName()`?  Or does it return something strange?